string_manipulation.rb

```diff
@@ -75,6 +82,9 @@ def reverse_words(my_words)
   return my_words
 end
 
+#Time complexity:  n * (n + n) = 2n^2 = O(n^2)  For every time we go through the outer loop, we go through two nested inner loop until we get to the end of the word before it can be reversed.  It is also affected by the size of the input i.e my_words.
+
+# Space complexity:  O(1), no new space is being created in memory
 
 #  I don't know why I couldn't get this to work
 # def reverse_words(my_words)
```

Comment:
On line 117, you are doing `j += -1` which will subtract 1 from j. Should this be adding 1 instead? Like so ` j += 1`. Same issue on lines 111 and 112. 
Once those are fixed, _I think_ this version of your implementation should also work.